### PR TITLE
kubeadm: use a suggested example in v1beta2/3 docs

### LINKED ISSUE #102767: Example kubeadm config has host addresses in the pod subnet
#### What happened:
I am trying to learn how to use kubeadm well and it looks like there may be a problem in the example configuration in https://github.com/kubernetes/kubernetes/blob/v1.21.1/cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go#L155-L253 .  It says `podSubnet: "10.100.0.1/24"`, but also uses address 10.100.0.1 for various host things like the kube-apiserver and the etcd server and the scheduler.  Also, interestingly, it puts a different address --- 10.100.1.1 --- in the apiserver's cert SANs.

#### What you expected to happen:
I expected to not be confused.

#### Environment:
- Kubernetes version (use `kubectl version`): 1.21.1
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Here is an example for dual-stack: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/dual-stack-support/.

#### Which issue(s) this PR fixes:

Fixes #102767

#### Special notes for your reviewer:
BTW, should we add dual-stack example here? @neolit123  Or keep it as is.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go [hunk 1]
```diff
@@ -203,8 +203,8 @@ limitations under the License.
 // 	    # certFile: "/etcd/kubernetes/pki/etcd/etcd.crt"
 // 	    # keyFile: "/etcd/kubernetes/pki/etcd/etcd.key"
 // 	networking:
-// 	  serviceSubnet: "10.96.0.0/12"
-// 	  podSubnet: "10.100.0.1/24"
+// 	  serviceSubnet: "10.96.0.0/16"
+// 	  podSubnet: "10.244.0.1/24"
```
Comment (SataQiu) [inline cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go:207]:
```suggestion
// 	  podSubnet: "10.244.0.0/24"
```

File: cmd/kubeadm/app/apis/kubeadm/v1beta3/doc.go [hunk 2]
```diff
@@ -204,10 +204,10 @@ limitations under the License.
 // 	    # certFile: "/etcd/kubernetes/pki/etcd/etcd.crt"
 // 	    # keyFile: "/etcd/kubernetes/pki/etcd/etcd.key"
 // 	networking:
-// 	  serviceSubnet: "10.96.0.0/12"
-// 	  podSubnet: "10.100.0.1/24"
+// 	  serviceSubnet: "10.96.0.0/16"
+// 	  podSubnet: "10.244.0.1/24"
```
Comment (SataQiu) [inline cmd/kubeadm/app/apis/kubeadm/v1beta3/doc.go:208]:
```suggestion
// 	  podSubnet: "10.244.0.0/24"
```

**Review by SataQiu — COMMENTED** [review]
Thanks @pacoxu 
LGTM, other than some nits!

**Review by SataQiu — APPROVED** [review]
/approve

**Comment by neolit123** [comment]
I think for dual stack users can navigate through the k8s.io website for more info. Having only single stack only in the kubeadm API serns fine to me.

/lgtm

**Pull request merged**